### PR TITLE
Grafana

### PR DESCRIPTION
**Pull request by miguelvr**
checklist:

- [x] run `make test` and `make lint`
- [x] test manually (i.e. build/push all images, restart operator, and re-deploy APIs)
- [x] update docs and add any new files to `summary.md` (view in [gitbook](https://cortex-labs.gitbook.io/staging/-MOmCGMADSRNQahK3Kox/) after merging)

### CODE REVIEW

File: pkg/types/clusterconfig/cluster_config_aws.go [hunk 1]
```diff
@@ -298,20 +298,6 @@ var CoreConfigStructFieldValidations = []*cr.StructFieldValidation{
...
-			Validator: validateImageVersion,
-		},
-	},
-	{
-		StructField: "ImageIstioPilot",
-		StringValidation: &cr.StringValidation{
-			Default:   "quay.io/cortexlabs/istio-pilot:" + consts.CortexVersion,
-			Validator: validateImageVersion,
-		},
-	},
```
Comment (deliahu) [inline pkg/types/clusterconfig/cluster_config_aws.go:314]: Should these be brought back?

Comment (miguelvr) [reply pkg/types/clusterconfig/cluster_config_aws.go:314]: it was repeated

File: pkg/operator/resources/realtimeapi/api.go [hunk 2]
```diff
@@ -37,6 +38,8 @@ import (
 	kcore "k8s.io/api/core/v1"
 )
 
+const _realtimeDashboardUID = "xvWFsZPGk"
```
Comment (deliahu) [inline pkg/operator/resources/realtimeapi/api.go:41]: Consider "apis"

File: pkg/operator/resources/realtimeapi/api.go [hunk 3]
```diff
@@ -273,18 +250,15 @@ func GetAPIByName(deployedResource *operator.DeployedResource) ([]schema.APIResp
...
-			Spec:     *api,
-			Status:   status,
-			Metrics:  metrics,
-			Endpoint: apiEndpoint,
-			//DashboardURL: dashboardURL,
+			Spec:         *api,
+			Status:       status,
+			Metrics:      metrics,
+			Endpoint:     apiEndpoint,
+			DashboardURL: dashboardURL,
```
Comment (deliahu) [inline pkg/operator/resources/realtimeapi/api.go:261]: Consider adding to batch APIs

**Review by deliahu — APPROVED** [review]

File: charts/templates/grafana.yaml [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+metadata:
+  name: grafana-storage
+  namespace: {{ .Release.Namespace }}
+spec:
+  storageClassName: ssd
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 2Gi
```
Comment (deliahu) [inline charts/templates/grafana.yaml:99]: Just out of curiosity, how did you determine that 2Gi is sufficient?

File: charts/templates/grafana.yaml [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+
+---
+
+apiVersion: apps/v1
+kind: StatefulSet
```
Comment (deliahu) [inline charts/templates/grafana.yaml:104]: It is ensured that this pod will be on the same node as Prometheus?

Comment (miguelvr) [reply charts/templates/grafana.yaml:104]: not at the moment, but I can add a podAffinity rule for that

File: charts/templates/grafana.yaml (hunk 4 above)
Comment (miguelvr) [reply charts/templates/grafana.yaml:99]: 2Gi should be plenty, as Grafana does not store much, just configs and its internal database for storing alerts, login info, etc...

**Review by deliahu — APPROVED** [review]

**Pull request merged**